src/requestor/routes/query.py

```diff
@@ -14,19 +14,32 @@
 from ..auth import Auth
 from ..config import config
 from ..models import Request as RequestModel
+from ..request_utils import flatten_query_params
 
 
 router = APIRouter()
 
 
-async def get_user_requests(username: str, active: bool = False) -> list:
+async def get_user_requests(
+    username: str, active: bool = False, filters: dict = {}
+) -> list:
+    """
+    Get all the requests made by current user.
+    If only active requests are needed then set active=True
+    Add filters if neccessary as a dictionary of {param : <List of values>} to get filtered results
+    """
     query = RequestModel.query.where(RequestModel.username == username)
     if active:
         query = query.where(
             RequestModel.status.notin_(
                 config["DRAFT_STATUSES"] + config["FINAL_STATUSES"]
             )
         )
+    for filter in filters:
+        query = RequestModel.query.where(
+            getattr(RequestModel, filter).in_(filters[filter])
```

Comment:
`for field, values in filters.items()` and then `values` instead of `filters[filter]` would be a bit easier to read.
or at least `for field in filters`, since a `for` over a dict returns the "field" key only, not the "filter" tuple